pandas/src/inference.pyx

```diff
@@ -686,9 +686,9 @@ def maybe_convert_numeric(object[:] values, set na_values,
                             raise ValueError('integer out of range')
                     else:
                         seen_float = True
-            except:
+            except (TypeError, ValueError) as e:
                 if not coerce_numeric:
-                    raise
+                    raise type(e)(e.message + ' in row {}'.format(i))
```

Comment:
Sorry to do a bit purist, but 'array index' sounds also a bit strange when you pass a Series and not an array. And given that just 'at index' can be confused with the actual index labels, maybe just 'at position ..' ?